components/style/media_queries.rs

```diff
@@ -243,19 +243,13 @@ pub fn parse_media_query_list(input: &mut Parser) -> MediaList {
     }
 
     let mut media_queries = vec![];
-    let mut found_invalid = false;
     loop {
         match input.parse_until_before(Delimiter::Comma, MediaQuery::parse) {
-            Ok(mq) => if !found_invalid {
+            Ok(mq) => {
                 media_queries.push(mq);
             },
-            Err(..) => if !found_invalid {
-                media_queries.clear();
+            Err(..) => {
                 media_queries.push(MediaQuery::never_matching());
```

Comment:
It doesn't preserve the expression, which matches gecko's behavior of serializing incorrect expressions as `not all`.